keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/UnhealthyRepository.kt

```diff
@@ -0,0 +1,28 @@
+package com.netflix.spinnaker.keel.persistence
+
+import org.slf4j.LoggerFactory
+import java.time.Duration
+
+/**
+ * Tracks resource health so that we can see how long a resource has been unhealthy for
+ */
+abstract class UnhealthyRepository() {
+
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  init {
+    log.info("Using ${javaClass.simpleName}")
+  }
+
+  /**
+   * Marks resource unhealthy, returns the duration it has been unhealthy for
```

Comment:
Combining the return with marking here is not super intuitive to me... What's the use case?

Reply:
I was trying to just make one function call. I'll remove this return value and make two calls.